File: go/microwave.go
package main

import (
	"time"
	"fmt"
	"flag"
	"strconv"
	"log"
)

const frequency = 10.0
const sleepTime = (time.Second / frequency)

func main() {
	flag.Parse();
	cookTime, err := strconv.Atoi(flag.Args()[0])
	if err != nil {
		log.Fatal("Please pass a valid number for cook time.")
	}
	cookTimeMinutes := cookTime / 60
	cookTimeSecondsDisplay := cookTime % 60
	fmt.Printf("COOK TIME: %02d:%02d\nBEEP!\n", cookTimeMinutes, cookTimeSecondsDisplay)
	reps := frequency * cookTime
	for i := 0; i < reps; i++ {
		fmt.Print("m")
		time.Sleep(sleepTime)
	}
	fmt.Println("\nBEEEP! BEEEP! BEEP!")
	fmt.Println("Food's done! :)")
}

File: rust/microwave.rs
use std::{env, thread, time};
use std::io::{self, Write};


fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        panic!("No cook time provided.");
    }
    let frequency = 10;
    let sleep_time = time::Duration::from_secs_f64(1.0 / frequency as f64);
    let cook_time_secs = match args[1].parse::<usize>() {
        Ok(usize) => usize,
        Err(_) => panic!("Invalid or missing cook time provided."),
    };
    let cook_time_mins_display = cook_time_secs / 60;
    let cook_time_secs_display = cook_time_secs % 60;
    println!("COOK TIME: {:02}:{:02}", cook_time_mins_display, cook_time_secs_display);
    let reps = cook_time_secs * frequency;
    println!("BEEP!");
    for _ in 0..reps {
        print!("m");
        io::stdout().flush().unwrap();
        thread::sleep(sleep_time);
    }
    println!("\nBEEEP! BEEEP! BEEEP!");
    println!("Food's done! :)");
}
